scripts/dist-git/remove_unused_sources

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/bash
+
+Usage() {
+    cat <<EOF
+Usage:
+    $0 <package_git_directory> <package_lookaside_directory>
+
+    Removes all tarballs in <package_lookaside_directory> except
+    for tarballs that are referenced by the latest commit of each
+    branch in <package_git_directory>.
+EOF
+}
+
+if [[ $# != 2 ]]; then
+    Usage
+    exit 1
+fi
+
+pkg_git_dir="$1"
+pkg_lookaside_dir="$2"
+
+if [ ! -d "$pkg_git_dir" ]; then
+    echo "$pkg_git_dir is not a valid directory."
+    exit 1
+fi
+if [ ! -d "$pkg_lookaside_dir" ]; then
+    echo "$pkg_lookaside_dir is not a valid directory."
+    exit 1
+fi
+
+pushd "$pkg_git_dir" > /dev/null
+
+whitelist=()
+
+# find sources that are referenced by the latest commit in any of the branches
+for branch in `git for-each-ref --format="%(refname:short)" refs/heads`; do
+    while read line; do
+        set -- $line
+        filename=$2
+        test $# -gt 2 && filename=$(echo "$filename" | sed -e "s/^(//" -e "s/)$//")
+        whitelist+=("$filename")
+    done < <(git show "$branch":sources)
+done
+
+popd > /dev/null
+
+# remove all sources that are not referenced
+for source_dir in "$pkg_lookaside_dir"/*; do
```

Comment:
This could work:
```bash
#! /bin/bash

pkg_lookaside_dir=/var/lib/dist-git/cache/lookaside/pkgs/praiskup/ping-temp/tar

# Read the old format directory (filename/md5sum/filenaem)
while read file; do
    old_IFS=$IFS
    IFS=/
    set -- $file
    IFS=$old_IFS
    # safety measure, if this is really the layout we expect the first and
    # third part matches
    test "$1" = "$3" || {
        echo "skipped $file"
        continue
    }
    echo unlink "$pkg_lookaside_dir/$file"
done < <( cd "$pkg_lookaside_dir" ; find -mindepth 3 -maxdepth 3 -type f  -printf '%P\n' )
```

Once anyone feels like the "new" format needs to be supported, it can easily be added.

Reply:
I added /var/tmp/test-crawl-script on stg dist-git for testing:
```
# /var/tmp/test-crawl-script /var/lib/dist-git/cache/lookaside/pkgs/praiskup/ping-temp/tarskipped tar-1.32.tar.xz/83e38700a80a26e30b2df054e69956e5/tar-1.32.tar.xz.DONT_DELETE
unlink /var/lib/dist-git/cache/lookaside/pkgs/praiskup/ping-temp/tar/tar-1.32.tar.xz/83e38700a80a26e30b2df054e69956e5/tar-1.32.tar.xz
```